scan-command/src/main/java/io/ballerina/scan/internal/ScanCmd.java

```diff
@@ -165,36 +175,96 @@ public void execute() {
             return;
         }
 
-        List<Issue> issues = projectAnalyzer.analyze(coreRules);
-        issues.addAll(projectAnalyzer.runExternalAnalyzers(externalAnalyzers));
-        if (!platforms.isEmpty() || platformTriggered) {
+        List<String> externalJarFilePaths = new ArrayList<>();
+        Map<String, PlatformPluginContext> platformContexts = new HashMap<>();
+        scanTomlFile.get().getPlatforms().forEach(platform -> {
+            if (platformTriggered && platforms.size() == 1 && platforms.contains(platform.name())) {
+                externalJarFilePaths.add(platform.path());
+                Map<String, String> platformArgs = platform.arguments().entrySet().stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, value -> value.getValue().toString()));
+                platformContexts.put(platform.name(), new PlatformPluginContextIml(
+                        (HashMap<String, String>) platformArgs, platformTriggered));
+            } else {
+                platforms.add(platform.name());
+                externalJarFilePaths.add(platform.path());
+                Map<String, String> platformArgs = platform.arguments()
+                        .entrySet()
+                        .stream()
+                        .collect(Collectors.toMap(Map.Entry::getKey, value -> value.getValue().toString()));
+                platformContexts.put(platform.name(), new PlatformPluginContextIml(
+                        (HashMap<String, String>) platformArgs, platformTriggered));
+            }
+        });
+
+        URLClassLoader ucl = loadPlatformPlugins(externalJarFilePaths);
+        ServiceLoader<StaticCodeAnalysisPlatformPlugin> scannerPlatformPlugins = ServiceLoader.load(
+                StaticCodeAnalysisPlatformPlugin.class, ucl);
+        scannerPlatformPlugins.forEach(staticCodeAnalysisPlatformPlugin -> {
+            if (platforms.contains(staticCodeAnalysisPlatformPlugin.platform())) {
+                PlatformPluginContext platformPluginContext = platformContexts.get(staticCodeAnalysisPlatformPlugin
+                        .platform());
+                staticCodeAnalysisPlatformPlugin.init(platformPluginContext);
+            }
+        });
+
+        includeRules.addAll(scanTomlFile.get().getRulesToInclude().stream().map(ScanTomlFile.RuleToFilter::id)
+                .toList());
+        excludeRules.addAll(scanTomlFile.get().getRulesToExclude().stream().map(ScanTomlFile.RuleToFilter::id)
+                .toList());
```

Comment:
Given these are not concurrent streams, I think we can factor these to avoid creating intermediate lists (in `toList()`). (Since we are have to fill the array sequentially any benefit we have from using vector instructions in `addAll` should be negated anyway.)
```suggestion
        scanTomlFile.get().getRulesToInclude().stream().map(ScanTomlFile.RuleToFilter::id).foreach(includeRules::add);
        scanTomlFile.get().getRulesToExclude().stream().map(ScanTomlFile.RuleToFilter::id).foreach(excludeRules::add);
```

Reply:
Check other places as well